examples/recording_service/pluggable_storage/c/CMakeLists.txt

```diff
@@ -0,0 +1,134 @@
+#
+# (c) 2019 Copyright, Real-Time Innovations, Inc.  All rights reserved.
+#
+#  RTI grants Licensee a license to use, modify, compile, and create derivative
+#  works of the Software.  Licensee has the right to distribute object form
+#  only for use with RTI products.  The Software is provided "as is", with no
+#  warranty of any type, including any warranty for fitness for any purpose.
+#  RTI is under no obligation to maintain or support the Software.  RTI shall
+#  not be liable for any incidental or consequential damages arising out of the
+#  use or inability to use the software.
+#
+cmake_minimum_required(VERSION 3.11)
+project (PluggableStorage)
+
+# Find RTI Connext dependencies
+set(CMAKE_MODULE_PATH ${CMAKE_MODULE_PATH} "${CONNEXTDDS_DIR}/resource/cmake")
+
+find_package(
+    RTIConnextDDS "6.0.0"
+    REQUIRED
+    COMPONENTS
+        core
+        routing_service
+)
+
+find_package(
+    RTICodeGenerator "3.0.0"
+    REQUIRED
+)
+
+# Generate type plug-in files for the storage type
+add_custom_command(
+    OUTPUT 
+        ${CMAKE_SOURCE_DIR}/HelloMsg.c 
+        ${CMAKE_SOURCE_DIR}/HelloMsgPlugin.c
+        ${CMAKE_SOURCE_DIR}/HelloMsgSupport.c
+    MAIN_DEPENDENCY ${CMAKE_SOURCE_DIR}/HelloMsg.idl
+    COMMAND ${RTICODEGEN} -language C -replace -d ${CMAKE_SOURCE_DIR} ${CMAKE_SOURCE_DIR}/HelloMsg.idl
+)
+
+# Define the library that will provide the storage writer plugin
+add_library(
+    FileStorageWriter MODULE
+    ${CMAKE_SOURCE_DIR}/FileStorageWriter.c
+    ${CMAKE_SOURCE_DIR}/FileStorageWriter.h
+)
+
+target_link_libraries(
+    FileStorageWriter
+    RTIConnextDDS::routing_service_infrastructure
+    RTIConnextDDS::c_api
+    ${CONNEXTDDS_EXTERNAL_LIBS}
+)
+
+target_include_directories(
+    FileStorageWriter
+    PUBLIC ${CMAKE_BINARY_DIR}
```

Comment:
The ``PUBLIC`` visibility is used when you are linking against the target (in this case, the ``FileStorageWriter`` library). 

You created the ``FileStorageWriter`` library as a ``MODULE`` library.

From the CMake documentation:
```
MODULE libraries are plugins that are not linked into other targets but may be loaded dynamically at runtime using dlopen-like functionality.
```

So, this means nobody will link against the ``FileStorageWriter`` library. For that reason, you should use ``PRIVATE`` visibility.

Reply:
I have not understood this correctly. This is actually a library that will not be linked into other targets but will effectively be dl-open-loaded by Recorder (the same as FileStorageReader will be dl-open-loaded by Replay and Converter), so I do not understand why MODULE doesn't fit?